src/main/java/com/gotocompany/firehose/sink/grpc/GrpcSinkFactory.java

```diff
@@ -26,22 +38,64 @@
  * using the configurations supplied and invoke {@see #create(Map < String, String > configuration, StatsDClient client)}
  * to obtain the GrpcSink sink implementation.
  */
+@Slf4j
 public class GrpcSinkFactory {
 
+    private static final String PERSIST_CERT_DIRECTORY_PATH = "/usr/local/share/ca-certificates";
+
+    private static File createCertFileFromBase64(String base64Cert) throws IOException {
+        byte[] decodedBytes = Base64.getDecoder().decode(base64Cert);
+        File certDirectory = new File(PERSIST_CERT_DIRECTORY_PATH);
+        if (!certDirectory.exists()) {
+            certDirectory.mkdirs();
+        }
+        File certFile = Files.createTempFile(certDirectory.toPath(), "client-cert", ".crt").toFile();
+        log.info("Created certificate file at {}", certFile.getAbsolutePath());
+        try (FileOutputStream fos = new FileOutputStream(certFile)) {
+            fos.write(decodedBytes);
+        }
+        return certFile;
+    }
+
+    private static SslContext buildClientSslContext(String base64Cert) {
+        try {
+            File certFile = createCertFileFromBase64(base64Cert);
```

Comment:
Just wondering whether can we skip the temporary file and directly pass the cert as `ByteArrayInputStream` instead. Maybe something like this 

`    private static SslContext buildClientSslContext(String base64Cert) {
        try {
            byte[] decodedBytes = Base64.getDecoder().decode(base64Cert);
            ByteArrayInputStream certInputStream = new ByteArrayInputStream(decodedBytes);
            SslContextBuilder sslContextBuilder = SslContextBuilder.forClient().trustManager(certInputStream);
            return GrpcSslContexts
                    .configure(sslContextBuilder)
                    .build();
        } catch (IOException e) {
            throw new RuntimeException("Failed to build SSL context due to an IO error while creating the certificate file.", e);
        }
    }`